src/deluge/gui/views/view.cpp

```diff
@@ -1194,17 +1194,19 @@ void View::modButtonAction(uint8_t whichButton, bool on) {
 
 void View::setModLedStates() {
 
-	bool itsTheSong = activeModControllableModelStack.getTimelineCounterAllowNull() == currentSong
-	                  || (!activeModControllableModelStack.timelineCounterIsSet()
-	                      && (getRootUI() == &sessionView || getRootUI() == &arrangerView));
+	bool itsTheSong =
+	    activeModControllableModelStack.getTimelineCounterAllowNull() == currentSong
```

Comment:
this is super unreadable, can you refactor to use a switch on the root UI instead of all the nested ifs ands and ors?